modules/memfault-firmware-sdk/include/memfault_lte_coredump_modem_trace.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+
+/**
+ * @brief Memfault LTE coredump modem trace initialization. Shall only be called once.
+ *	  Ths function sets up a CDR (Custom Data Recording) source for modem traces.
+ *
+ * @retval -EALREADY if the module has already been initialized.
+ * @retval 0 if the modem trace backend was successfully initialized.
+ */
+int memfault_lte_coredump_modem_trace_init(void);
+
+/**
+ * @brief Prepare modem trace data for upload
+ *
+ * @retval -ENOTSUP if the current modem trace backend is not supported.
+ * @retval -ENODATA if there are no modem traces to send.
+ * @retval 0 if the modem trace data is ready for upload.
+ *
+ * @returns Negative error if an error occurred preparing the modem trace.
```

Comment:
`@returns` is likely redundant given the error codes are listed just above?

Reply:
The apis underneath can return other negative error codes looks like. So I added is as a catch all.